src/script_rs/src/main.rs

```diff
@@ -0,0 +1,64 @@
+use std::time::Instant;
+use rand::{Rng, SeedableRng};
+use rand_xoshiro::Xoshiro256PlusPlus;
+
+fn dotindex(x: &[f64]) -> f64 {
+    x.iter().enumerate().fold(0.0, |acc, (i, v)| v.mul_add(i as f64, acc))
+}
+
+fn f(n: i32, m: i32) -> f64 {
+    let n = n as usize;
+    let mut y: f64 = 0.0;
+    let mut x: Vec<f64> = Vec::with_capacity(n);
+    x.resize(n, 0.0);
+    let mut rng = Xoshiro256PlusPlus::from_rng(rand::thread_rng()).unwrap();
+
+    for _ in 0..m {
+        rng.fill(&mut x[..]);
+        x.sort_unstable_by(|a, b| a.partial_cmp(b).unwrap());
+        y += dotindex(&x);
+    }
+    let n = n as f64;
+    y / m as f64 / (n * n)
+}
+
+fn g(n: i32, m: i32) -> f64 {
+    let n = n as usize;
+    let mut y: f64 = 0.0;
+    let mut x: Vec<f64> = Vec::with_capacity(n);
+    x.resize(n, 0.0);
+    let mut rng = Xoshiro256PlusPlus::from_rng(rand::thread_rng()).unwrap();
+
+    for _ in 0..m {
+        rng.fill(&mut x[..]);
+        y += dotindex(&x);
```

Comment:
```suggestion
        rng.fill(&mut x[..]);

        y += dotindex(&x);
```
Leave a blank line here for stylistic consistency with other implementations and to make parallels with `f` more clear.

Reply:
Updated with `rustfmt`, then added line.